library/Reporting/Web/Forms/ReportForm.php

```diff
@@ -124,6 +157,34 @@ protected function assemble()
         }
     }
 
+    public function validate()
+    {
+        parent::validate();
+        if (! $this->isValid) {
+            return false;
+        }
+
+        if ($this->id || $this->getPopulatedValue('remove')) {
```

Comment:
Is this really needed? The remove button should have the `formnovalidate` attribute set. Is `validate()` called in that case?